src/NServiceBus.Persistence.DynamoDB/Outbox/DynamoOutboxConfigurationExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace NServiceBus.Persistence.DynamoDB;
+
+using System;
+using Configuration.AdvancedExtensibility;
+using Outbox;
+
+/// <summary>
+/// Outbox configuration extensions for DynamoDB persistence.
+/// </summary>
+public static class DynamoOutboxConfigurationExtensions
+{
+    /// <summary>
+    /// Customize the configuration of the table used by the outbox persistence.
+    /// </summary>
+    public static OutboxSettings UseTable(this OutboxSettings outboxSettings, Func<TableConfiguration, TableConfiguration> tableConfiguration)
```

Comment:
really not a big fan of this but I didn't want to make this a plain `TableConfiguration` parameter since that would require the user to always new up a new instance while in many cases you probably just want to adjust one setting on table configuration.

I also didn't want to make this an Action<TableConfiguration> to still easily allow to replace the default table config easily.

An alternative might be to split this into two different extension methods to support both cases?

Reply:
Why is newing up that configuration object which comes with sensible defaults such a problem? Isn't a lambda expression far more difficult and potentially cumbersome to use? Not every developer might be familiar with lambdas but they are with newing up objects.

Reply:
I was mostly thinking about the `UseSharedTable` API which currently replaces the default config on both saga and outbox. It then wouldn't be possible to easily tweak something on the outbox specifically (while that's easy to do with the Saga config), although I'm also not sure that is a very likely use case.